applications/serial_lte_modem/src/nrf52_dfu/dfu_host.c

```diff
@@ -0,0 +1,629 @@
+	NRF_DFU_OP_RECEIPT_NOTIF_SET = 0x02,     /* Set receipt notification */
+	NRF_DFU_OP_CRC_GET           = 0x03,     /* Request CRC of selected object */
+	NRF_DFU_OP_OBJECT_EXECUTE    = 0x04,     /* Execute selected object */
+	NRF_DFU_OP_OBJECT_SELECT     = 0x06,     /* Select object */
+	NRF_DFU_OP_MTU_GET           = 0x07,     /* Retrieve MTU size */
+	NRF_DFU_OP_OBJECT_WRITE      = 0x08,     /* Write selected object */
+	NRF_DFU_OP_PING              = 0x09,     /* Ping */
+	NRF_DFU_OP_HARDWARE_VERSION  = 0x0A,     /* Retrieve hardware version */
+	NRF_DFU_OP_FIRMWARE_VERSION  = 0x0B,     /* Retrieve firmware version */
+	NRF_DFU_OP_ABORT             = 0x0C,     /* Abort the DFU procedure */
+	NRF_DFU_OP_RESPONSE          = 0x60,     /* Response */
+	NRF_DFU_OP_INVALID           = 0xFF
+};
+
+/**
+ * @brief DFU operation result code.
+ */
+enum nrf_dfu_result_t {
+	/* Invalid opcode */
+	NRF_DFU_RES_CODE_INVALID                 = 0x00,
+	/* Operation successful */
+	NRF_DFU_RES_CODE_SUCCESS                 = 0x01,
+	/* Opcode not supported */
+	NRF_DFU_RES_CODE_OP_CODE_NOT_SUPPORTED   = 0x02,
+	/* Missing/invalid parameter value */
+	NRF_DFU_RES_CODE_INVALID_PARAMETER       = 0x03,
+	/* Not enough memory for the data object */
+	NRF_DFU_RES_CODE_INSUFFICIENT_RESOURCES  = 0x04,
+	/* Data object does not match the firmware and hardware requirements,
+	 * the signature is wrong, or parsing the command failed
+	 */
+	NRF_DFU_RES_CODE_INVALID_OBJECT          = 0x05,
+	/* Not a valid object type for a Create request */
+	NRF_DFU_RES_CODE_UNSUPPORTED_TYPE        = 0x07,
+	/* The state of the DFU process does not allow this operation */
+	NRF_DFU_RES_CODE_OPERATION_NOT_PERMITTED = 0x08,
+	/* Operation failed */
+	NRF_DFU_RES_CODE_OPERATION_FAILED        = 0x0A,
+	/* Extended error. The next byte of the response contains the error
+	 * code of the extended error (see @ref nrf_dfu_ext_error_code_t
+	 */
+	NRF_DFU_RES_CODE_EXT_ERROR               = 0x0B,
+};
+
+/**
+ * @brief @ref NRF_DFU_OP_OBJECT_SELECT response details.
+ */
+struct nrf_dfu_response_select_t {
+	uint32_t offset;                    /* Current offset */
+	uint32_t crc;                       /* Current CRC */
+	uint32_t max_size;                  /* Maximum size of selected object */
+};
+
+/**
+ * @brief @ref NRF_DFU_OP_CRC_GET response details.
+ */
+struct nrf_dfu_response_crc_t {
+	uint32_t offset;                    /* Current offset */
+	uint32_t crc;                       /* Current CRC */
+};
+
+#define MIN(a, b) (((a) < (b)) ? (a) : (b))
+
+static uint8_t  ping_id;
+static uint16_t prn;
+static uint16_t mtu;
+
+static uint8_t send_data[REQ_DATA_SIZE_MAX];
+static uint8_t receive_data[RSP_DATA_SIZE_MAX];
+
+/* global functions defined in different resources */
+int dfu_drv_tx(const uint8_t *p_data, uint16_t length);
+int dfu_drv_rx(uint8_t *p_data, uint32_t max_len, uint32_t *p_real_len);
+
+static uint16_t get_u16_le(const uint8_t *p_data)
```

Comment:
There's `sys_get_le16()` in `sys/byteorder.h`.

Reply:
Change to use Zephyr API